airbyte-integrations/connectors/source-mailchimp/source_mailchimp/components.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.sources.declarative.auth.declarative_authenticator import DeclarativeAuthenticator
+from airbyte_cdk.sources.declarative.auth.token import BasicHttpAuthenticator, BearerAuthenticator
+from airbyte_cdk.sources.declarative.extractors.record_filter import RecordFilter
+from airbyte_cdk.sources.declarative.requesters.http_requester import HttpRequester
+from airbyte_cdk.sources.declarative.requesters.request_options.interpolated_request_options_provider import (
+    InterpolatedRequestOptionsProvider,
+    RequestInput,
+)
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+
+
+@dataclass
+class MailChimpRequester(HttpRequester):
```

Comment:
can you describe why this is needed? it's difficult to judge whether is should truly be custom behavior without stating the desired behavior and why it cannot be supported natively by the CDK

Reply:
So, mailchimp declares custom [api structure](https://mailchimp.com/developer/marketing/docs/fundamentals/#api-structure)

> There are a few ways to find your data center. It’s the first part of the URL you see in the[ API keys section](https://us1.admin.mailchimp.com/account/api/) of your account; if the URL is https://us6.mailchimp.com/account/api/, then the data center subdomain is us6. It’s also appended to your API key in the form key-dc; if your API key is `0123456789abcdef0123456789abcde-us6`, then the data center subdomain is us6. And finally, if you’re connecting via OAuth 2, you can find the data center associated with the token via the OAuth Metadata endpoint; for more information, see the [OAuth guide](https://mailchimp.com/developer/marketing/guides/access-user-data-oauth-2/#implement-the-oauth-2-workflow-on-your-server).

The only reason for this custom component is to get the `data center` prefix for Oauth authenticator.

Reply:
Why is this a custom requester instead of something like a config migrator?